.github/scripts/sync_labels.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+import sys
+import requests
+import json
+from github import Github
+
+try:
+    github_token = os.environ["GITHUB_TOKEN"]
+except KeyError:
+    print("Please set the 'GITHUB_TOKEN' environment variable")
+    sys.exit(1)
+
+
+def parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--repository', type=str, required=True,
+                        help='Github repository name (e.g., scylladb/scylladb)')
+    parser.add_argument('--number', type=int, required=True,
+                        help='Pull request number to sync labels from linked issue')
+    parser.add_argument('--is_issue', action='store_true', help='   Determined if label change is in Issue or not')
+    parser.add_argument('--label_action', type=str, choices=['opened', 'labeled', 'unlabeled'], required=True, help='Sync labels action')
+    return parser.parse_args()
+
+
+def get_pr_request_body(repo, number):
+    url = f"https://api.github.com/repos/{repo}/pulls/{number}"
+    headers = {
+        'Authorization': f'token {github_token}',
+        'Accept': 'application/vnd.github.v3+json'
+    }
+    response = requests.get(url, headers=headers)
+    if response.ok:
+        pr_details = response.json()
+        return pr_details['body']
+    else:
+        print("Failed to fetch PR details. Status code:", response.status_code)
+
+
+def get_linked_issues(pr_body):
+    pattern = re.compile(r'Fixes:? (?:#|https.*?/issues/)(\d+)', re.IGNORECASE)
+    matches = re.findall(pattern, pr_body)
+    if not matches:
+        raise RuntimeError("No regex matches found in the body!")
+    issue_numbers = []
+    for match in matches:
+        issue_numbers.append(match)
+        print(f"Found issue number: {match}")
+    return issue_numbers
+
+
+def get_linked_prs(repo, issue_number, token):
+    query = f"repo:{repo} is:pr is:open linked:issue {issue_number}"
+    url = f"https://api.github.com/search/issues?q={query}"
+    headers = {"Authorization": f"Bearer {token}"}
+    response = requests.get(url, headers=headers)
+    if response.status_code == 200:
+        prs = response.json().get('items', [])
+        return prs
+    else:
+        raise Exception(f"GitHub API returned {response.status_code}: {response.text}")
+
+
+def sync_labels(issue_numbers, number, repo, label_action):
+    print('Sync labels from Issue to linked pull request')
+    github = Github(github_token)
+    pull_request_labels = [label.name for label in github.get_repo(repo).get_pull(int(number)).get_labels()]
```

Comment:
maybe this one should be inside the:
if label_action == 'opened':

Reply:
yes you are right , moving